src/OrchardCore.Modules/OrchardCore.Contents/Security/ContentTypeAuthorizationHandler.cs

```diff
@@ -37,9 +38,19 @@ protected override async Task HandleRequirementAsync(AuthorizationHandlerContext
 
             if (contentItem != null)
             {
-                if (OwnerVariationExists(requirement.Permission) && HasOwnership(context.User, contentItem))
+                if (contentItem.Owner != null) // If we want to evaluate a specific content item authorization
                 {
-                    permission = GetOwnerVariation(requirement.Permission);
+                    if (OwnerVariationExists(requirement.Permission) && HasOwnership(context.User, contentItem))
+                    {
+                        permission = GetOwnerVariation(requirement.Permission);
+                    }
+                }
+                else // If we want to evaluate if a user has authorization on a ContentType globally
```

Comment:
@sebastienros Here is what I added for allowing to not need to inject the IHttpContextAccessor everywhere we need to evaluate if the user can create content items of a certain content type. There was only 2 places in the code where we need this so far and people made mistake of not passing the ContentItem.Owner so it was failing to evaluate. Here, we also would evaluate the same value against each other because the "patch" would be to call the authorization service like this : AuthorizationService.AutorizeAsync(IUser, IPermission, _httpContext.User.Identity.Name). And you can see on line 129 that we evaluate ContentItem.Owner = User.Identity.Name ... which is always going to return true ...